File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    let answer = []
    const CARPET_SIZE = brown + yellow
    
    for (let i = 3; i < CARPET_SIZE; i++) {
        let width = CARPET_SIZE / i
        let height = i
        
        if (width < height) continue

        if ((width - 2) * (height - 2) === yellow) {
            answer[0] = width
            answer[1] = height
        }
    }
    return answer
}

File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    total_tiles = brown + yellow 
    for height in range(3, total_tiles + 1):
        if total_tiles % height == 0:
            width = total_tiles // height

            if width < height:
                continue
            if (width - 2) * (height - 2) == yellow:
                answer = [width, height]
                break
    return answer